src/include/processor/operator/order_by/top_k.h

```diff
@@ -0,0 +1,187 @@
+#pragma once
+
+#include <queue>
+
+#include "function/binary_function_executor.h"
+#include "processor/operator/sink.h"
+#include "sort_state.h"
+
+namespace kuzu {
+namespace processor {
+
+using vector_select_comparison_func =
+    std::function<bool(common::ValueVector&, common::ValueVector&, common::SelectionVector&)>;
+
+struct TopKScanState {
```

Comment:
Leave a TODO for me to move this into mapper

Reply:
I don't think we can move the initialization to mapper.
We need have access to the sortedKeyBlock. If the sorting hasn't started, we don't have access to the sortedKeyBlock.